analyzer/evmverifier/sourcify/client.go

```diff
@@ -86,35 +86,78 @@ func (s *SourcifyClient) callAPI(ctx context.Context, method string, url *url.UR
 // Note: This uses the free, public server API. If it turns out to be unreliable, we could use the repository API (vis IPFS proxy) instead, e.g.:
 // http://ipfs.default:8080/ipns/repo.sourcify.dev/contracts/full_match/23294
 func (s *SourcifyClient) GetVerifiedContractAddresses(ctx context.Context, runtime common.Runtime) (map[ethCommon.Address]VerificationLevel, error) {
-	// Fetch verified contract addresses.
+	// Build map of addresses.
+	addresses := make(map[ethCommon.Address]VerificationLevel)
+
+	// Fetch fully verified contract addresses.
+	// See https://docs.sourcify.dev/docs/full-vs-partial-match/
 	u := *s.serverUrl
-	u.Path = path.Join(u.Path, "files/contracts", sourcifyChains[s.chain][runtime])
-	body, err := s.callAPI(ctx, http.MethodGet, &u)
+	u.Path = path.Join(u.Path, "files/contracts/full", sourcifyChains[s.chain][runtime])
+	fullyVerified, err := s.GetAllAddressPages(ctx, &u)
 	if err != nil {
-		return nil, fmt.Errorf("failed to fetch verified contract addresses: %w (%s)", err, u.String())
+		return nil, err
 	}
-
-	// Parse response.
-	var response struct {
-		Full    []ethCommon.Address `json:"full"`
-		Partial []ethCommon.Address `json:"partial"` // See https://docs.sourcify.dev/docs/full-vs-partial-match/
+	for _, addr := range fullyVerified {
+		addresses[addr] = VerificationLevelFull
 	}
-	if err := json.Unmarshal(body, &response); err != nil {
-		return nil, fmt.Errorf("failed to parse verified contract addresses: %w (%s)", err, u.String())
+	// Fetch partially verified contract addresses.
+	u = *s.serverUrl
+	u.Path = path.Join(u.Path, "files/contracts/any", sourcifyChains[s.chain][runtime])
+	allContracts, err := s.GetAllAddressPages(ctx, &u)
+	if err != nil {
+		return nil, err
 	}
-
-	// Build map of addresses.
-	addresses := make(map[ethCommon.Address]VerificationLevel)
-	for _, addr := range response.Full {
-		addresses[addr] = VerificationLevelFull
+	for _, addr := range allContracts {
+		if _, exists := addresses[addr]; !exists {
+			addresses[addr] = VerificationLevelPartial
+		}
 	}
-	for _, addr := range response.Partial {
-		addresses[addr] = VerificationLevelPartial
+
+	return addresses, nil
+}
+
+func (s *SourcifyClient) GetAllAddressPages(ctx context.Context, u *url.URL) ([]ethCommon.Address, error) {
+	page := 0
+	addresses := []ethCommon.Address{}
+	for {
+		// https://sourcify.dev/server/api-docs/#/Repository/get_files_contracts_any__chain_
+		q := u.Query()
+		q.Set("page", string(page))
```

Comment:
lint says this doesn't give the string of the number :skull:

Reply:
ya i guess it's more like `chr()`; updated to use `fmt.Sprintf` tho